torch/distributions/multivariate_normal.py

```diff
@@ -0,0 +1,154 @@
+import math
+from numbers import Number
+
+import torch
+from torch.autograd import Variable
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all, lazy_property
+
+
+def _get_batch_shape(bmat, bvec):
+    """
+    Given a batch of matrices and a batch of vectors, compute the combined `batch_shape`.
+    """
+    try:
+        vec_shape = torch._C._infer_size(bvec.shape, bmat.shape[:-1])
+    except RuntimeError:
+        raise ValueError("Incompatible batch shapes: vector {}, matrix {}".format(bvec.shape, bmat.shape))
+    return torch.Size(vec_shape[:-1])
+
+
+def _batch_mv(bmat, bvec):
+    """
+    Performs a batched matrix-vector product, with an arbitrary batch shape.
+    """
+    batch_shape = bvec.shape[:-1]
+    event_dim = bvec.shape[-1]
+    bmat = bmat.expand(batch_shape + (event_dim, event_dim))
+    if batch_shape != bmat.shape[:-2]:
+        raise ValueError("Batch shapes do not match: matrix {}, vector {}".format(bmat.shape, bvec.shape))
+    bvec = bvec.unsqueeze(-1)
+    
+    # using `torch.bmm` is surprisingly clunky... only works when `.dim() == 3`
```

Comment:
Before you send upstream, consider replacing with something more diplomatic :wink:
> conform to torch.bmm which requires .dim() == 3